### ISSUE TITLE
The GoLang back-end should support specifying module names for Dafny modules

### ISSUE THREAD

**Issue by keyboardDrummer** [dafny-lang/dafny#4925]
The GoLang back-end should support specifying a GoLang module name for a specific Dafny module. This Dafny module can be either part of the sources of the current project, or part of dependencies specified through `--library` that are not translated to GoLang but can be referenced through import statements.

CLI syntax: `dafny translate -t go dfyconfig.toml --go-module Some.Dafny.Module github.com/Org/RepoName Some.Other.DModule github.com/Org2/RepoName2`

```
# dfyconfig.toml
includes = ["src/contains/some/dafny/module.dfy"]

[options]
go-module = [
  ["Some.Dafny.Module", "github.com/Org/RepoName"],
  ["Some.Other.DModule", "github.com/Org2/RepoName2"]
]

library = ["lib/contains/some/other/dafny/module.doo"]

**Comment by ShubhamChaturvedi7** [comment]
We would want to map Dafny projects to Go modules, not Dafny modules to Go modules. Today, Dafny modules are rightly mapped to Go packages. Changing this mapping to Go modules means a multi-module Dafny project (without using the library flag) can end up as multiple projects (go modules) in Go world. Word Module has different meaning in Dafny and Go, and we shouldn't try mapping them 1:1.

**Comment by ShubhamChaturvedi7** [comment]
Similar issue is also present in Python world too.

**Comment by keyboardDrummer** [comment]
> Word Module has different meaning in Dafny and Go, and we shouldn't try mapping them 1:1.

I agree. This issue is suggesting mapping them X:1, a Dafny module and all its nested modules, to 1 Go module.

> We would want to map Dafny projects to Go modules, not Dafny modules to Go modules.

Why? I don't see a need for that restriction. To me a Dafny project is a less well defined concept than a Dafny module.

I'm guessing your scenario is one where the Dafny project has no 'root module' in which all other modules are nested, so you can not map the root module to a Go module to achieve the same affect as mapping your Dafny project to a Go module.

I will update the 'outer module' flag so it affectively works as adding a root module in which all your other modules are nested, so you can add a root module to your project by using that, without having to update any of your sources.

**Comment by keyboardDrummer** [comment]
> Similar issue is also present in Python world too.

Can you elaborate?
